x/bank/src/keeper/mod.rs

```diff
@@ -287,16 +92,20 @@ impl<SK: StoreKey, PSK: ParamsSubspaceKey, AK: AuthKeeper<SK, M>, M: Module>
     pub fn init_genesis<DB: Database>(
         &self,
         ctx: &mut InitContext<'_, DB, SK>,
-        genesis: GenesisState,
+        mut balances: Vec<Balance>,
+        params: BankParams,
+        denom_metadata: Vec<Metadata>,
     ) {
-        // TODO:
-        // 1. cosmos SDK sorts the balances first
-        // 2. Need to confirm that the SDK does not validate list of coins in each balance (validates order, denom etc.)
-        // 3. Need to set denom metadata
-        self.bank_params_keeper.set(ctx, genesis.params);
+        // 1. cosmos SDK sorts the balances first - Make sure that rust ordering gives same result
+        // 2. Need to confirm that the SDK does not validate list of coins in each balance (validates order, denom etc.) - Yes it does and our Coins type did it
+        // 3. Need to set denom metadata - dedicated cmd for it
+        self.bank_params_keeper.set(ctx, params);
+
+        // Make sure that this 100% same as in cosmos(generally it should)
```

Comment:
This looks like a TODO?

Reply:
I'm more confident in this rather than don't. So don't think this is todo